Makefile

```diff
@@ -2,15 +2,44 @@ VENV = venv
 PYTEST = $(PWD)/$(VENV)/bin/py.test
 
 # These targets are not files
-.PHONY: install sandbox docs coverage lint messages compiledmessages css clean sandbox_image
+.PHONY: build_sandbox clean compile_translations coverage css docs extract_translations help install install-python \
+ install-test install-js lint release retest sandbox_clean sandbox_image sandbox test todo venv
 
-install:
+help: ## Display this help message
```

Comment:
I think the PR generally looks good, but I am not sure about the value of this `help` command, and the complexity of the implementation makes me nervous. Does it add enough value over just reading the make file to justify maintaining it?

Also, I'm assuming from trying to parse it that it requires perl as a system dependency?

Reply:
ah, i get your point. and yes it requires perl. 
i will remove this.

Reply:
@specialunderwear Yeah i agree, those are useful. I have used grep to get strings but again i think this will fail on windows. @craigloftus what you think, we can skip this if you think its not worth it.